egs/librispeech/ASR/prepare_gigaspeech.py

```diff
@@ -0,0 +1,293 @@
+            from distributed import Client
+
+            with plz.setup_cluster() as cluster:
+                cluster.scale(80)
+                yield Client(cluster)
+            return
+    except:
+        pass
+    # No need to return anything - compute_and_store_features
+    # will just instantiate the pool itself.
+    yield None
+
+
+def locate_corpus(*corpus_dirs):
+    for d in corpus_dirs:
+        if os.path.exists(d):
+            return d
+    print(
+        "Please create a place on your system to put the downloaded Librispeech data "
+        "and add it to `corpus_dirs`"
+    )
+    sys.exit(1)
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument(
+        "--num-jobs",
+        type=int,
+        default=min(5, os.cpu_count()),
+        help="Number of parallel jobs.",
+    )
+    parser.add_argument(
+        "--subset",
+        type=str,
+        default="XS",
+        help="Select the GigaSpeech subset (XS|S|M|L|XL)",
+    )
+    parser.add_argument(
+        "--context-window",
+        type=float,
+        default=0.0,
+        help="Training cut duration in seconds. "
+        "Use 0 to train on supervision segments without acoustic context, with variable cut lengths; "
+        "number larger than zero will create multi-supervisions cuts with actual acoustic context. ",
+    )
+    parser.add_argument(
+        "--context-direction",
+        type=str,
+        default="center",
+        help="If context-window is 0, does nothing. "
+        "If it's larger than 0, determines in which direction (relative to the supervision) "
+        "to seek for extra acoustic context. Available values: (left|right|center|random).",
+    )
+    parser.add_argument(
+        "--precomputed-features",
+        type=str2bool,
+        default=True,
+        help="Should we pre-compute features and store them on disk or not. "
+        "It is recommended to disable it for L and XL splits as the pre-computation "
+        "might currently consume excessive memory and time -- use on-the-fly feature "
+        "extraction in the training script instead.",
+    )
+    return parser
+
+
+# Similar text filtering and normalization procedure as in:
+# https://github.com/SpeechColab/GigaSpeech/blob/main/toolkits/kaldi/gigaspeech_data_prep.sh
+
+
+def normalize_text(
+    utt: str,
+    punct_pattern=re.compile(r"<(COMMA|PERIOD|QUESTIONMARK|EXCLAMATIONPOINT)>"),
+    whitespace_pattern=re.compile(r"\s\s+"),
+) -> str:
+    return whitespace_pattern.sub(" ", punct_pattern.sub("", utt))
+
+
+def has_no_oov(
+    sup: SupervisionSegment, oov_pattern=re.compile(r"<(SIL|MUSIC|NOISE|OTHER)>")
+) -> bool:
+    return oov_pattern.search(sup.text) is None
+
+
+def main():
+    args = get_parser().parse_args()
+    dataset_parts = [args.subset, "DEV", "TEST"]
+    if args.subset in ["L", "XL"]:
+        assert is_module_available("pyarrow"), (
```

Comment:
This doesn't look right to me; `pyarrow` is long gone from Lhotse, which version of the recipe is that? I don't see this code in https://github.com/k2-fsa/snowfall/blob/master/egs/gigaspeech/asr/simple_v1/prepare.py

Reply:
Will modify it to the master one.

I directly copy the file from this link https://github.com/k2-fsa/snowfall/pull/230/files 
The website defaultly show "the pyarrow version", which exactly not the same with the one in the master branch.

Reply:
If you search pyarrow from previous link, you would got the "pyarrow" version prepare.py .
https://github.com/k2-fsa/snowfall/pull/230/files#diff-10ff23c2be6cdc66d1461d8ff40b8acee607ab6c2e5d734c7ce03b65765289c3R143

I always thought github would show the latest version, but it seems not so.

Reply:
I think I made follow-up PRs where I synchronized the changes with Lhotse and also improved GigaSpeech recipe.

Reply:
I know what happened; 
I was told that https://github.com/k2-fsa/snowfall/pull/230 is Piotr's implementation about gigaspeech.

But some files related are also modified in pr https://github.com/k2-fsa/snowfall/pull/238

That's why the older version is refered.